python/pfs/drp/stella/readLineList.py

```diff
@@ -75,8 +76,9 @@ def filterByLampElements(self, lines, lampElements):
         """
         keep = []
         for element in lampElements:
-            keep += [ll for ll in lines if ll.description.startswith(element)]
-        self.log.info("Filtered line lists for element/species: %s", ",".join(sorted(element)))
+            keep += [ll for ll in lines if re.match(f'^{element}[IVX]*$', ll.description)]
```

Comment:
This is in a loop, so let's compile the regex first.

Reply:
I don't think your regex is quite right, as it allows selecting `HgII` when I specify `HgI`.
```
>>> element = "HgI"
>>> description = "HgII"
>>> re.match(f'^{element}[IVX]*$', description)
<re.Match object; span=(0, 4), match='HgII'>
```